### PR TITLE
update fullstack branching docs

### PR DESCRIPTION
**Pull request by Jay2113**
#### Description of changes:
Updated the following pages under the deployment (formerly deploy&host) section:
- fullstack branch deployments
- secrets and environment vars
- share resources across branches
- separate frontend and backend teams
- monorepo setup
- fullstack previews
- custom pipelines
- cross account deployments

#### Related GitHub issue #, if available:
N/A

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx [hunk 1]
```diff
@@ -95,13 +95,13 @@ npx ampx generate outputs --app-id BACKEND-APP --branch main --format dart --out
 </Callout>
 
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id BACKEND-APP --branch main --out-dir app/src/main/res/raw
+npx ampx generate outputs --app-id BACKEND-APP-ID --branch main --format json-mobile --out-dir app/src/main/res/raw
```
Comment (josefaidt) [inline src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx:98]:
```suggestion
npx ampx generate outputs --app-id <your-backend-amplify-app-id> --branch main --out-dir app/src/main/res/raw
```

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx [hunk 2]
```diff
@@ -78,13 +78,13 @@ The first app, `my-shared-backend`, will be the only app that updates changes to
...
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --branch main --app-id BACKEND-APP
+npx ampx generate outputs --branch main --app-id BACKEND-APP-ID
 ```
 </InlineFilter>
 
 <InlineFilter filters={['flutter']}>
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id BACKEND-APP --branch main --format dart --out-dir lib
+npx ampx generate outputs --app-id BACKEND-APP-ID --branch main --format dart --out-dir lib
```
Comment (josefaidt) [inline src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx:87]:
```suggestion
npx ampx generate outputs --app-id <your-backend-amplify-app-id> --branch main --format dart --out-dir lib
```

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx [hunk 3]
```diff
@@ -95,13 +95,13 @@ npx ampx generate outputs --app-id BACKEND-APP --branch main --format dart --out
...
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id BACKEND-APP --branch main --out-dir app/src/main/res/raw
+npx ampx generate outputs --app-id BACKEND-APP-ID --branch main --format json-mobile --out-dir app/src/main/res/raw
 ```
 </InlineFilter>
 
 <InlineFilter filters={['swift']}>
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id BACKEND-APP
+npx ampx generate outputs --app-id BACKEND-APP-ID --format json-mobile
```
Comment (josefaidt) [inline src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx:104]:
```suggestion
npx ampx generate outputs --app-id <your-backend-amplify-app-id> --branch main
```

File: src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx [hunk 4]
```diff
@@ -91,17 +91,17 @@ npx ampx generate outputs --app-id <your-amplify-app-id> --branch main --out-dir
...
 <InlineFilter filters={['flutter']}>
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id <your-amplify-app-id> --branch main --format dart --out-dir lib
+npx ampx generate outputs --app-id <your-amplify-app-id> --branch <your-git-branch-name> --format dart --out-dir lib
 ```
 </InlineFilter>
 <InlineFilter filters={['swift']}>
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id <your-amplify-app-id>
+npx ampx generate outputs --app-id <your-amplify-app-id> --branch <your-git-branch-name> --format=json-mobile
```
Comment (josefaidt) [inline src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx:104]:
```suggestion
npx ampx generate outputs --app-id <your-amplify-app-id> --branch <your-git-branch-name> 
```

File: src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx [hunk 5]
```diff
@@ -91,17 +91,17 @@ npx ampx generate outputs --app-id <your-amplify-app-id> --branch main --out-dir
 </Callout>
 
 ```bash title="Terminal" showLineNumbers={false}
-npx ampx generate outputs --app-id <your-amplify-app-id> --branch main --out-dir app/src/main/res/raw
+npx ampx generate outputs --app-id <your-amplify-app-id> --branch <your-git-branch-name> --format json-mobile --out-dir app/src/main/res/raw
```
Comment (josefaidt) [inline src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx:94]:
```suggestion
npx ampx generate outputs --app-id <your-amplify-app-id> --branch <your-git-branch-name> --out-dir app/src/main/res/raw
```

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx (hunk 1 above)
Comment (Jay2113) [reply src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx]: Done!

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx (hunk 2 above)
Comment (Jay2113) [reply src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx]: Done!

File: src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx (hunk 3 above)
Comment (Jay2113) [reply src/pages/[platform]/deploy-and-host/fullstack-branching/monorepos/index.mdx]: Done!

File: src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx (hunk 4 above)
Comment (Jay2113) [reply src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx]: Done!

File: src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx (hunk 5 above)
Comment (Jay2113) [reply src/pages/[platform]/deploy-and-host/fullstack-branching/branch-deployments/index.mdx]: Done!

**Review by josefaidt — APPROVED** [review]

**Review by aspittel — APPROVED** [review]

**Pull request merged**